src/state/ducks/programsDuck.js

```diff
@@ -1,23 +1,112 @@
-const programsATs = {};
+import { axiosAuth } from '../../api/axiosAuth';
 
-export const programsActions = {};
+//Action Types
+const GET_PROGRAM_START = ' GET_PROGRAM_START';
+const GET_PROGRAM_SUCCESS = ' GET_PROGRAM_SUCCESS';
+const GET_PROGRAM_FAIL = ' GET_PROGRAM_FAIL';
+const GET_PROGRAM_RESOLVE = ' GET_PROGRAM_RESOLVE';
+const GET_ALLPROGRAMS_START = 'GET_ALLPROGRAMS_START';
+const GET_ALLPROGRAMS_SUCCESS = 'GET_ALLPROGRAMS_SUCCESS';
+const GET_ALLPROGRAMS_FAIL = 'GET_ALLPROGRAMS_FAIL';
+const GET_ALLPROGRAMS_RESOLVE = 'GET_ALLPROGRAMS_RESOLVE';
 
+//Action Creators
+export const programsActions = {
+  //GET All Programs Action
+  getAllProgramsThunk: () => dispatch => {
+    dispatch({ type: GET_ALLPROGRAMS_START });
+
+    axiosAuth()
+      .get('/programs/programs')
+      .then(res => {
+        console.log(res);
+        dispatch({ type: GET_ALLPROGRAMS_SUCCESS, payload: res.data });
+      })
+      .catch(err => {
+        dispatch({ type: GET_ALLPROGRAMS_FAIL, payload: err.message });
+      })
+      .finally(() => {
+        dispatch({ type: GET_ALLPROGRAMS_RESOLVE });
+      });
+  },
+
+  //GET Individual Program Action
+  getProgramThunk: programId => dispatch => {
+    dispatch({ type: GET_PROGRAM_START });
+
+    axiosAuth()
+      .get(`/programs/program/${programId}`)
+      .then(res => {
+        console.log(res);
+        dispatch({ type: GET_PROGRAM_SUCCESS, payload: res.data });
+      })
+      .catch(err => {
+        dispatch({ type: GET_PROGRAM_FAIL, payload: err.message });
+      })
+      .finally(() => {
+        dispatch({ type: GET_PROGRAM_RESOLVE });
+      });
+  },
+};
+
+//Initial Slice of State
 const programsInitialState = {
   programs: [],
-  isFetching: false,
-  error: '',
   program: {
-    id: '',
+    programId: '',
     programName: '',
     programType: '',
     programDescription: '',
     courses: [],
-    // users: [],
   },
+  status: 'idle',
+  error: '',
 };
 
+//Reducers
 const programsReducer = (state = programsInitialState, action) => {
   switch (action.type) {
+    //Get All Programs Reducers
+    case GET_ALLPROGRAMS_START:
+      return { ...state, status: 'pending' };
+
+    case GET_ALLPROGRAMS_SUCCESS:
+      return {
+        ...state,
+        programs: [...state.programs, action.payload],
+        status: 'success',
+      };
+    case GET_ALLPROGRAMS_FAIL:
+      return { ...state, status: 'error', error: action.payload };
+
+    case GET_ALLPROGRAMS_RESOLVE:
+      return { ...state, status: 'idle' };
+
+    //Get Individual Program Reducers
+    case GET_PROGRAM_START:
+      return { ...state, status: 'pending' };
+
+    case GET_PROGRAM_SUCCESS:
+      return {
```

Comment:
don't forget you can also destructure `action.payload` before `return` statement to save on keystrokes! 

```
case GET_PROGRAM_SUCCESS:
    const {programId, programName, programType, programDescription, courses} = action.payload;
    return {
      ...state,
      program: {
        ...state.program,
        programId: programId,  // could be verbose
        programName,           // could use key-val shorthand (this does the same thing as the line above... key `programName` will have value `<value of programName variable>`
        programType,
        programDescription,
        courses: [...state.program.courses, courses]  // this one couldn't use shorthand cuz we're spreading existing state.program.courses
      }      
    }
```

Reply:
Ahhh yes of course, fantastic suggestion. Will make this change!